nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -292,6 +293,7 @@ def _determine_log_key(self, data_config, dataloader_idx, metric_name, mode):
 
     def fwd_bwd_step(self, dataloader_iter, batch_idx, forward_only):
         batch = next(dataloader_iter)
+        batch = {k: v for k, v in batch.items() if isinstance(v, torch.Tensor)}
```

Comment:
can you comment why this is needed? cc @aklife97

Reply:
I pass `reference_texts` in the batch which is not a torch.Tensor and will get errors when doing `data_iter = get_iterator_k_split(batch, get_num_microbatches())`

Reply:
In the batch, we may have other metadata column from dataset. So I simply select those are tensors to pass to `get_iterator_k_split()`. If I don't do so, we may see errors for element not be tensor.

Reply:
could you comment in the code as well

Reply:
please comment in the code as @ekmb mentioned.

Reply:
Sure!